### PR TITLE
New API cleanup

### PR DESCRIPTION
**Pull request by JDWarner**
Functionally the most important change is naming `WeightedConsequent` -> `WeightedTerm`, and `FuzzyVariableTerm` became `Term`. Given how these worked and what they required, this reflects their actual function.

Lots more docstrings and style improvements. This addresses in large part #77, but there is still room for some improvement. Improved `view` for Terms so the legend reflects the bold, active membership function.

Also, the tests should now pass! 

What I consider blocking before 0.2 release:

- [ ] Ability to repeatedly `.compute` using a single simulation object
- [ ] Check for overly sparse system, warn the user, and catch these cases so more informative error messages are raised

What do you think, @jsexauer?

### CODE REVIEW

File: skfuzzy/control/visualization.py [hunk 1]
```diff
@@ -1,39 +1,63 @@
...
-        -------
+        assert (isinstance(fuzzy_var, FuzzyVariable) or
+                isinstance(fuzzy_var, Term))
+
+        # self.term allows us to know if this is a Term quickly, later
+        self.term = None
+        if isinstance(fuzzy_var, Term):
+            self.term = fuzzy_var.label
+            self.fuzzy_var = fuzzy_var.parent_variable
+        else:
```
Comment (jsexauer) [inline skfuzzy/control/visualization.py]: I would use elif isinstance(fuzzy_var, FuzzyVariable), raise a `ValueError` in the else, and remove the assert.

**Comment by jsexauer** [comment]
Looks great!  Thanks.

File: skfuzzy/control/visualization.py (hunk 1 above)
Comment (JDWarner) [inline skfuzzy/control/visualization.py]: Good call, I'm on it.

**Comment by JDWarner** [comment]
Merging, continuing work in another PR.

**Pull request merged**